### DISCUSSION TITLE
Why does LTeX check in a different language than ltex.language?

### DISCUSSION THREAD

**Discussion by mykappa** [valentjn/vscode-ltex#245] [Q&A] [answered]
I have a problem when using LTeX: 

Although I have specified `"ltex.enabled": true, "ltex.language": "en-US",` in my general VScode settings, I still get `GERMAN_SPELLER_RULE` spell checker warnings  in my English LaTeX text. My system language (Windows) is German. How can I stop LTeX from doing this? I just want it to do the spell check in English...

Thanks in advance for any suggestions!

**Comment by valentjn** [comment, marked as answer]
Your system language should not be related to this. There are some reasons why LT<sub>E</sub>X checks in a different language than what `ltex.language` says in the user configuration, [see the docs](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html):

- [You set `ltex.language` in the `settings.json` of a workspace or a workspace folder.](https://code.visualstudio.com/docs/getstarted/settings)
- [You use magic comments.](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html#magic-comments)
- [You use the babel package.](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html#multilingual-latex-documents-with-the-babel-package)

If it's none of these, then you should [open a bug report](https://valentjn.github.io/vscode-ltex/docs/contributing-code-issues.html#how-to-report-bugs) with the necessary info.

**Comment by mykappa** [comment]
Thank you for the quick reply!

Indeed, babel seems to be the culprit. I started with a template where babel was included. After some adjustments (and several compilation runs) I deleted the command, since the document language will be English. However, LTeX seems to somehow retain the language settings and continues to display German spelling errors even when I reload VScode and delete all auxiliary LaTeX files. 

I don't know if this is a bug or if I'm missing something...

**Reply by valentjn** [reply]
If the problem persists when reloading the VS Code window, I suspect you still have some babel commands in there. Do a global search for the [four commands/environments listed in the docs](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html#multilingual-latex-documents-with-the-babel-package) and *delete* (not comment out) all occurrences. (Currently, LT<sub>E</sub>X is not smart enough to see if a babel command is commented out. ~~I guess this should be fixed soon.~~ This is fixed in the next release.)

LT<sub>E</sub>X only uses the T<sub>E</sub>X code of the file in which the errors are displayed. No other T<sub>E</sub>X files, no auxiliary files.

**Reply by mykappa** [reply]
Thanks for the clarification. I actually found that one of the commands mentioned was just commented out. Now it works. Great that LTeX will be able to recognize when babel is commented out in the future. 

And in general, thank you for providing such a great tool and quick support!

**Reply by valentjn** [reply]
Fix for commented out babel commands released in 9.0.0.

**Comment by Unfaehig** [comment]
I still struggle with this topic. I get that there are the ways of defining it with magic comments / babble. This works great in my main tex. However i split up my tex files. And in each external tex it will fall back to `en-US`. This is my OS language. But not the document lang nor what i set in coc setting :(

coc-settings.json
```json
{
  "languageserver": {
    "ltex": {
      "filetypes": [ "tex" ],
      "language": "ngerman"
    }
  }
}
```
